src/stdlib_constants.fypp

```diff
@@ -0,0 +1,66 @@
+#:include "common.fypp"
+module stdlib_constants
+    !! Constants
+    use stdlib_kinds, only: sp, dp, qp
+    use stdlib_codata, only: SPEED_OF_LIGHT_IN_VACUUM, &
+                             VACUUM_ELECTRIC_PERMITTIVITY, &
+                             VACUUM_MAG_PERMEABILITY, &
+                             PLANCK_CONSTANT, &
+                             NEWTONIAN_CONSTANT_OF_GRAVITATION, &
+                             STANDARD_ACCELERATION_OF_GRAVITY, &
+                             ELEMENTARY_CHARGE, &
+                             MOLAR_GAS_CONSTANT, &
+                             FINE_STRUCTURE_CONSTANT, &
+                             AVOGADRO_CONSTANT, &
+                             BOLTZMANN_CONSTANT, &
+                             STEFAN_BOLTZMANN_CONSTANT, &
+                             WIEN_WAVELENGTH_DISPLACEMENT_LAW_CONSTANT, &
+                             RYDBERG_CONSTANT, &
+                             ELECTRON_MASS, &
+                             PROTON_MASS, &
+                             NEUTRON_MASS, &
+                             ATOMIC_MASS_CONSTANT
+    private
+
+    ! mathematical constants
+    #:if WITH_QP
+    real(qp), parameter, public :: PI_qp    = 3.1415926535897932384626433832795028841971_qp !! PI quadruple precision
+    #:else
+    real(dp), parameter, public :: PI_dp    = 3.1415926535897932384626433832795028841971_dp !! PI double precision
+    #:endif
+    real(sp), parameter, public :: PI_sp    = real(pi_dp, sp) !! PI single precision
+
+    
+    ! Physical constants
+    real(dp), parameter, public :: c = SPEED_OF_LIGHT_IN_VACUUM%value !! Speed of light in vacuum
```

Comment:
I like the fact that these constants are only stored in 64-bit precision: most of them are accurate enough nowadays that 32-bit accuracy is not enough. But, at the same time, I've never seen anyone that is more than 15-digit accurate. May be worth checking though.